test/ered_SUITE.erl

```diff
@@ -67,8 +69,15 @@ wait_for_consistent_cluster() ->
                                 ered_client:stop(Pid),
                                 SlotMap
                         end(P) || P <- ?PORTS],
+            %% Get the amount of nodes in the first reply. The reply format has improved between
+            %% Redis 6 and 7, but to handle Redis 6 we count each expected port in the reply.
+            NoOfNodes = length([X || X <- [lists:member(integer_to_binary(P),
+                                                        binary:split(list_to_binary(hd(AllNodes)), <<"\n">>, [global]))
+                                           || P <- ?PORTS],
+                                     X =:= true]),
+
             case length(lists:usort(AllNodes)) of
-                1 ->
+                1 when NoOfNodes == length(?PORTS) ->
                     true;
                 _ when N > 0 ->
                     timer:sleep(500),
```

Comment:
Here, if we would want to check the contents of the CLUSTER SLOTS output, after checking that they're all identical, we could do

```Erlang
            case lists:usort(AllNodes) of
                [Nodes] ->
                    %% ... Check the contents of Nodes here ...
                    true;
                _NotAllIdentical ->
                    timer:sleep(500),
```